tests/aggregation/tests.py

```diff
@@ -2158,6 +2199,155 @@ def test_having_with_no_group_by(self):
+                "Mamma and Pappa's Books;Amazon.com;Books.com",
+            ),
+            (F("original_opening"), "Mamma and Pappa's Books;Amazon.com;Books.com"),
+            ("original_opening", "Mamma and Pappa's Books;Amazon.com;Books.com"),
+            ("-original_opening", "Books.com;Amazon.com;Mamma and Pappa's Books"),
+            (
+                Concat("original_opening", Value("@")),
+                "Mamma and Pappa's Books;Amazon.com;Books.com",
+            ),
+            (
+                Concat("original_opening", Value("@")).desc(),
+                "Books.com;Amazon.com;Mamma and Pappa's Books",
+            ),
+        )
+        for order_by, expected_output in order_by_test_cases:
+            with self.subTest(order_by=order_by, expected_output=expected_output):
+                values = Store.objects.aggregate(
+                    stringagg=StringAgg("name", delimiter=Value(";"), order_by=order_by)
+                )
+                self.assertEqual(values, {"stringagg": expected_output})
+
+    def test_string_agg_filter(self):
+        values = Book.objects.aggregate(
+            stringagg=StringAgg(
+                "name",
+                delimiter=Value(";"),
+                filter=Q(name__startswith="P"),
+            )
+        )
+
+        expected_values = {
+            "stringagg": "Practical Django Projects;"
+                         "Python Web Development with Django;Paradigms of Artificial "
+                         "Intelligence Programming: Case Studies in Common Lisp",
+        }
+        self.assertEqual(values, expected_values)
+
+    @skipUnlessDBFeature("supports_aggregate_order_by_clause")
+    def test_string_agg_jsonfield_order_by(self):
+        values = Book.objects.aggregate(
+            stringagg=StringAgg(
+                KeyTextTransform("lang", "print_info"),
+                delimiter=Value(","),
+                order_by=KeyTextTransform("lang", "print_info"),
+                output_field=CharField(),
+            ),
+        )
+        self.assertEqual(values, {"stringagg": "en,fr"})
+
+    def test_string_agg_filter_in_subquery(self):
+        """
+        This test is based on tests taken from existing PostgreSQL specific tests and
+        kept to avoid regressions as StringAgg is ported to the shared database module.
+        """
+
+        aggregate = StringAgg(
+            "authors__name",
+            delimiter=Value(";"),
+            filter=~Q(authors__name__startswith="J")
+        )
+        subquery = (
+            Book.objects.filter(
+                pk=OuterRef("pk"),
+            )
+            .annotate(agg=aggregate)
+            .values("agg")
+        )
+        values = list(
+            Book.objects.annotate(
+                agg=Subquery(subquery),
+            )
+            .exclude(agg__isnull=True)
+            # Different engines treat null STRING_AGG differently, so excluding it for
+            # consistency.
```

Comment:
Could you elaborate on that?

I seems that SQLite, Postgres, MySQL, and Oracle all return `NULL` which should translate to `None`?

1. [SQLite](https://dbfiddle.uk/fzC4JzJm)
2. [MySQL](https://dbfiddle.uk/jf5-JeSq)
3. [Postgres](https://dbfiddle.uk/62y1WY4y)
4. [Oracle](https://dbfiddle.uk/e0gP_WKJ)

Did you possibly run into issues because of a lack of ordering on the returned value and the fact some backends default to `NULLS LAST` instead of `NULLS FIRST`?

If that's the case you should use `assertQuerySetEqual(..., ordered=False)` or specify an `order_by(OrderBy('agg', nulls_last=True))`.

I think it'd be good to include at least one tests that ensures that `None` is returned on an empty not-NULL set.

Reply:
You are right, I was reading the test results wrong. I went with `assertQuerySetEqual(..., ordered=False)` to resolve this.